python/ray/projects/scripts.py

```diff
@@ -145,6 +145,113 @@ def load_project_or_throw():
             "`ray project validate` to inspect the error.")
 
 
+class SessionRunner(object):
+    """Class for setting up a session and executing commands in it."""
+
+    def __init__(self, command, args, shell):
+        """Initialize session runner and try to parse the command arguments.
+
+        Args:
+            command (str, optional): Command from the project definition's
+                commands section to run, if any.
+            args (list): Arguments for the command to run.
+            shell (bool): If true, command is a shell command that should be
+                run directly.
+        Raises:
+            click.ClickException: This exception is raised if any error occurs.
+        """
+        self.project_definition = load_project_or_throw()
+
+        # We try to parse the command here so we can fail before the cluster
+        # is started in case the command is malformed.
+        if shell:
+            self.command_to_run = command
```

Comment:
I think it actually might make more sense to pass in the command to run to `SessionRunner.execute_command` instead of the constructor. This would also let us reuse the same `SessionRunner` for multiple commands.

Reply:
Yeah, good idea!